lib/puppetscript.rb

```diff
@@ -30,14 +30,24 @@ def self.generate_user_array(users)
 
   def self.node_file_correct?(vm_name, contents)
     result =    contents.lines[0].chomp.eql?('class node_' + replace_dashes_with_underscores(vm_name) + ' {')
-    result &&=  contents.lines[1].start_with?('        $admins = [')
-    result &&=  contents.lines[2].start_with?('        $users = [')
-    result &&=  contents.lines[4].chomp.eql?('        realize(Accounts::Virtual[$admins], Accounts::Sudoroot[$admins])')
-    result &&=  contents.lines[5].chomp.eql?('        realize(Accounts::Virtual[$users])')
-    result &&=  contents.lines[6].chomp.eql?('}')
+    result &&=  contents.lines[1].strip.start_with?('$admins = [')
+    result &&=  contents.lines[2].strip.start_with?('$users = [')
+    result &&=  contents.lines[4].strip.eql?('realize(Accounts::Virtual[$admins], Accounts::Sudoroot[$admins])')
+    result &&=  contents.lines[5].strip.eql?('realize(Accounts::Virtual[$users])')
+    result &&=  contents.lines[6].strip.eql?('}')
     result
   end
 
+  def self.read_node_string(script)
+    values = { admins: [], users: [] }
+
+    admins = script.lines[1][/\[.*?\]/].delete('"[]').split(', ')
```

Comment:
We could make this more generic by relying solely on a regex to extract the user names, regardless of which line they are in, e.g. `\$admins\s*=\s*\[["|'](\S+)["|']\]` , see: https://rubular.com/r/4AJ5PUkYflmzcK

Reply:
I've added this comment to https://github.com/hpi-swt2/vm-portal/issues/514